datatypes/src/raster/operations/scaling.rs

```diff
@@ -0,0 +1,631 @@
+use crate::raster::{
+    data_type::DefaultNoDataValue, EmptyGrid, Grid, GridOrEmpty, GridOrEmpty2D, GridSize,
+    MapElements, MapElementsParallel, RasterTile2D,
+};
+use num_traits::{CheckedAdd, CheckedDiv, CheckedMul, CheckedSub};
+use std::ops::{Add, Div, Mul, Sub};
+
+/// scales the value with `(value - offset_by) / scale_with`.
```

Comment:
I would have expected value * scale + offset tbh. wouldn't this here be more like "unscale" or normalize=

Reply:
i used the GDAL terms from: https://gdal.org/programs/gdal_translate.html

Reply:
On the page it says

> -unscale[](https://gdal.org/programs/gdal_translate.html#cmdoption-gdal_translate-unscale)
>
>    Apply the scale/offset metadata for the bands to convert scaled values to unscaled values. It is also often necessary to reset the output datatype with the [-ot](https://gdal.org/programs/gdal_translate.html#cmdoption-gdal_translate-ot) switch.

isn't that what you're doing in your `scale` method? That's why I thought it should be called unscaled, but maybe I'm looking at it wrong.

Reply:
For GDAL scaled data is not the real data. I also think that is easy to misunderstand. See the "-scale" option in gdal_edit: https://gdal.org/programs/gdal_edit.html#cmdoption-scale


"Scale and Offset are generally used together. For example, scale and offset might be used to store elevations in a unsigned 16bit integer file with a precision of 0.1, and starting from -100. True values would be calculated as: true_value = (pixel_value * scale) + offset"